File: 1614-maximum-nesting-depth-of-the-parentheses/1614-maximum-nesting-depth-of-the-parentheses.cpp
class Solution {
public:
    int maxDepth(string s) {
     int count=0,Depth=0;

     for(int i=0;i<s.length();i++){
         if(s[i]=='(' ){
             count++;
             Depth=max(count,Depth);
         }
         else if(s[i]==')' && count>=1)
         count--;
     }   
     return Depth;
    }
};

File: 1614-maximum-nesting-depth-of-the-parentheses/1614-maximum-nesting-depth-of-the-parentheses.py
class Solution:
    def maxDepth(self, s: str) -> int:
        count=0
        depth=0
        length=len(s)

        for i in range(0,length):
            if s[i]=='(':
                count=count+1
                depth=max(depth,count)
            elif s[i]==')' and count>=1:
                count=count-1    
        
        return depth